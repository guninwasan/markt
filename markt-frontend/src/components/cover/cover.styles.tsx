import styled from "styled-components";

const MainContainer = styled.div`
  display: flex;
  height: 100vh;
`;

const CoverContainer = styled.div`
  display: flex;
  width: 55vw;
  height: 100vh;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  background: #b6b6b6;
  box-shadow: 0px 4px 4px 0px rgba(0, 0, 0, 0.25);
  overflow: hidden;
`;

const Title = styled.h1`
  font-size: 2.5rem;
`;

const Subtitle = styled.p`
  font-size: 1.5rem;
  text-align: center;
`;

const RightContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 45vw;
  padding: 2rem;
`;

export { MainContainer, CoverContainer, Title, Subtitle, RightContainer };
