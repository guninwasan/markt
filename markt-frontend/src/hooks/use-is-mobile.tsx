import { useEffect, useState } from "react";

const MOBILE_BREAKPOINT = 770;

const useIsMobile = () => {
  const [isMobile, setIsMobile] = useState<null | boolean>(null);

  useEffect(() => {
    const handleResize = () => {
      setIsMobile(window.innerWidth <= MOBILE_BREAKPOINT);
    };

    window.addEventListener("resize", handleResize);

    return () => {
      window.removeEventListener("resize", handleResize);
    };
  }, []);

  return { isMobile };
};

export { useIsMobile };
