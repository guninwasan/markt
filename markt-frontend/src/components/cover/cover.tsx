import React from "react";
import {
  MainContainer,
  CoverContainer,
  RightContainer,
  Subtitle,
  Title,
} from "./cover.styles";
import { useIsMobile } from "../../hooks/use-is-mobile";

type CoverProps = {
  children: React.ReactNode;
  title: string;
};

const Cover = ({ children, title }: CoverProps) => {
  const { isMobile } = useIsMobile();

  return (
    <MainContainer>
      {!isMobile && (
        <CoverContainer>
          <Subtitle>
            Your Campus.
            <br />
            Your Marketplace.
          </Subtitle>
        </CoverContainer>
      )}

      <RightContainer>
        <Title>{title}</Title>
        {children}
      </RightContainer>
    </MainContainer>
  );
};

export { Cover };
