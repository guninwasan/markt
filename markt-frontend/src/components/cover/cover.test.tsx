import React from "react";
import { render, screen } from "@testing-library/react";
import { Cover } from "./cover";
import { useIsMobile } from "../../hooks/use-is-mobile";

jest.mock("../../hooks/use-is-mobile");

const mockUseIsMobile = useIsMobile as jest.Mock;

describe("Cover", () => {
  const title = "Test Title";
  const children = <div>Test Children</div>;

  beforeEach(() => {
    mockUseIsMobile.mockClear();
  });

  it("should render the title and children", () => {
    mockUseIsMobile.mockReturnValue({ isMobile: false });

    render(<Cover title={title}>{children}</Cover>);

    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.getByText("Test Children")).toBeInTheDocument();
  });

  it("should render the subtitle when not on mobile", () => {
    mockUseIsMobile.mockReturnValue({ isMobile: false });

    render(<Cover title={title}>{children}</Cover>);

    expect(screen.getByText(/Your Campus./)).toBeInTheDocument();
    expect(screen.getByText(/Your Marketplace./)).toBeInTheDocument();
  });

  it("should not render the subtitle when on mobile", () => {
    mockUseIsMobile.mockReturnValue({ isMobile: true });

    render(<Cover title={title}>{children}</Cover>);

    expect(screen.queryByText("Your Campus.")).not.toBeInTheDocument();
    expect(screen.queryByText("Your Marketplace.")).not.toBeInTheDocument();
  });

  it("should match snapshot", () => {
    mockUseIsMobile.mockReturnValue({ isMobile: false });

    const { container } = render(<Cover title={title}>{children}</Cover>);

    expect(container).toMatchSnapshot();
  });
});
