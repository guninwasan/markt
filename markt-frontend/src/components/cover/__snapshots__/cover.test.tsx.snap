// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cover should match snapshot 1`] = `
<div>
  <div
    class="sc-blHHSb cgxrDM"
  >
    <div
      class="sc-gtLWhw eCKWsw"
    >
      <h2
        class="sc-dntaoT chJTIi"
      >
        Your Campus.
        <br />
        Your Marketplace.
      </h2>
    </div>
    <div
      class="sc-egkSDF eHyuTo"
    >
      <h2
        class="sc-fAUdSK djnvGh"
      >
        Test Title
      </h2>
      <div>
        Test Children
      </div>
    </div>
  </div>
</div>
`;
